lib/Backend/Lower.cpp

```diff
@@ -2896,7 +2896,23 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
         case Js::OpCode::StatementBoundary:
             // This instruction is merely to help convey source info through the IR
             // and eventually generate the nativeOffset maps.
-            break;
+#if DBG_DUMP && DBG
+            // If we have a JITStatementBreakpoint, then we should break on this statement
+            {
+                uint32 statementIndex = instr->AsPragmaInstr()->m_statementIndex;
+                if (Js::Configuration::Global.flags.StatementDebugBreak.Contains(instr->m_func->GetFunctionNumber(), statementIndex))
+                {
+                    IR::Instr* tempinstr = instr;
+                    // go past any labels, and then add a debug breakpoint
+                    while (tempinstr->m_next != nullptr && tempinstr->m_next->m_opcode == Js::OpCode::Label)
```

Comment:
I don't believe so; tempinstr is initialized to instr, which, if null, would cause issues much earlier in the surrounding function, and any values which might be assigned to tempinstr are first checked for nullity. I've added an assert in the new version though.